### ISSUE TITLE
VAE configuration issue

### ISSUE THREAD

**Issue by spezialspezial** [huggingface/diffusers#6753]
### Describe the bug

It seems vae configuration might be broken in recent commits (~7f58a76f). Even though I can verify the correct config contents are read in I still end up with the wrong vae config. Most notably the scale_factor for SDXL models is that of SD1.5 resulting in washed out generations. I suspect the error to be somewhere in create_diffusers_vae_model_from_ldm but not sure at the moment. Just me? Any ideas?

### Reproduction

1) Instantiate SDXL pipe from single file. 
2) Check your vae scaling factor. If it's not 0.13025 there might be an issue

### System Info

diffusers 0.26.0.dev0 7f58a76f, Linux, Python 3.10.12 with variadic function support, torch 2.1.2+cu121 + cuda 12.1

### Who can help?

@sayakpaul

**Comment by spezialspezial** [comment]
looks like the VAE always ends up with the default argument value from `create_diffusers_vae_model_from_ldm` and the yaml config scale_factor item is never used. See `create_vae_diffusers_config`

**Comment by sayakpaul** [comment]
Cc: @DN6

**Comment by spezialspezial** [comment]
sure thing
```
pipe = StableDiffusionXLPipeline.from_single_file("YOUR_PATH/sd_xl_base_1.0_0.9vae.safetensors")
print(pipe.vae.config.scaling_factor)
```

**Comment by DN6** [comment]
Should be fixed with #6779

**Comment by spezialspezial** [comment]
Awesome, much appreciated!
